internal/alert_exporter/alert_sender.go

```diff
@@ -0,0 +1,127 @@
+package alert_exporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type AlertSender struct {
+	log      *logrus.Logger
+	hostname string
+	port     uint
+}
+
+func NewAlertSender(log *logrus.Logger, hostname string, port uint) *AlertSender {
+	return &AlertSender{
+		log:      log,
+		hostname: hostname,
+		port:     port,
+	}
+}
+
+func (a *AlertSender) SendAlerts(checkpoint *AlertCheckpoint) error {
+	err := a.sendAlerts(checkpoint.Updated)
+	if err != nil {
+		return err
+	}
+
+	a.cleanupAlerts(checkpoint)
+	return nil
+}
+
+type AlertmanagerAlert struct {
+	Labels       map[string]string `json:"labels"`
+	Annotations  map[string]string `json:"annotations,omitempty"`
+	StartsAt     time.Time         `json:"startsAt"`
+	EndsAt       time.Time         `json:"endsAt,omitempty"`
+	GeneratorURL string            `json:"generatorURL,omitempty"`
+}
+
+// Send a list of alerts in batches to Alertmanager
+func (a *AlertSender) sendAlerts(alerts []*AlertInfo) error {
+	const batchSize = 100
+	alertmanagerAlerts := make([]AlertmanagerAlert, 0, len(alerts))
+
+	for _, alert := range alerts {
+		// Construct the AlertmanagerAlert from your AlertInfo
+		alertmanagerAlert := AlertmanagerAlert{
+			Labels: map[string]string{
+				"alertname": alert.Reason,
+				"resource":  alert.ResourceName,
```

Comment:
Should we include the resource kind and not only the name?
The event types not always identify the kind of resource.